io/plugins/eu.esdihumboldt.hale.io.json/src/eu/esdihumboldt/hale/io/json/JsonInstanceWriter.java

```diff
@@ -40,9 +46,18 @@ public boolean isCancelable() {
 	protected IOReport execute(ProgressIndicator progress, IOReporter reporter)
 			throws IOProviderConfigurationException, IOException {
 		progress.begin("Generating JSON", ProgressIndicator.UNKNOWN);
+
+		InstanceToJson instanceToJson = new InstanceToJson(true);
```

Comment:
> Ideally you would make it an optional constructor argument of this class
1. We don't have `optional arguments` in Java. Do you mean that I should create a default construct with no params and initialise the property to create a valid `GeoJson` to `false`? 
- I think the constructor that takes a boolean parameter is already a clean way to handle this. 

> GeoJSONInstanceWriter can just extend it and only needs to set it to true in the constructor.
2. Why should `GeoJSONInstanceWriter extends InstanceToJson` class? If we extend like this then we cannot extend `AbstractInstanceWriter` unless we would like to create another abstract class that will extend either `InstanceToJson` or `AbstractInstanceWriter` and then extending this `AbstractAbstractinstanceWriter` class into `GeoJSONInstanceWriter`. 
- I believe by `extend`, you meant to create a class variable like `InstanceToJson instanceToJson` and initialise this variable in the default constructor of `GeoJSONInstanceWriter` class with default value as `true`. This can also be achieved by the constructor that takes the boolean argument `public InstanceToJson(boolean geoJson) {.... `.

Reply:
> Ideally you would make it an optional constructor argument of this class

With "this class" I mean `JsonInstanceWriter`.
With "optional constructor argument" I meant basically adding a constructor with a boolean argument to `JsonInstanceWriter`, optional because we need to retain the default constructor.

> Why should GeoJSONInstanceWriter extends InstanceToJson class?

I meant for `GeoJSONInstanceWriter` to extend `JsonInstanceWriter`.

Reply:
Changes done to:
1. made `useGeoJsonFeatures` as an optional constructor argument defaulting it to `false`
2. extended `GeoJSONInstanceWriter ` with `JsonInstanceWriter`